File: 2021/day01/main.js
let fs = require("fs");

// let arr = fs.readFileSync('input.txt').toString('UTF8').split('\n').map(Number);
let arr = fs.readFileSync('in.txt').toString('UTF8').split('\n').map(Number);

console.log("read %d lines", arr.length);
console.log("first: %d, last: %d\n", arr[0], arr[arr.length-1]);

// pretty straightforward

A:
var last = arr[0];
let count = 0;
for (i of arr) {
	if (i > last) { last = i; count++ }
	last = i;
}
console.log(count);

B:
count = 0;
last = arr[0] + arr[1] + arr[2];
for (let i = 2; i < arr.length; i++) {
	sum = arr[i-2] + arr[i-1] + arr[i];
	if (sum > last) { count++ }
	last = sum;
}

console.log(count)

File: 2021/day01/main.swift

import Foundation

let filename = "in.txt"

let arr = try String(contentsOfFile: "in.txt")
print(arr)
let nums = arr.split(separator:"\n").map{Int($0)!}
print(nums)

var last = nums[0]
var sum = 0
for n in nums {
	if n>last {
		sum += 1
	}
	last = n
}
print(sum) // 1696
print()

last = nums[2]
sum = 0
for (i,_) in nums.enumerated() {
	if i<3 { continue }
	if (nums[i]+nums[i-1]+nums[i-2])>last {
		sum += 1
	}
	last = nums[i]+nums[i-1]+nums[i-2]
}
print(sum) // 1737

